src/Playwright/Core/Locator.cs

```diff
@@ -338,4 +342,249 @@ private async Task WithElementAsync(Func<IElementHandle, float?, Task> callback,
             },
             timeout).ConfigureAwait(false);
     }
+
+    public ILocator GetByAltText(string text, LocatorGetByAltTextOptions options = null)
+        => ((ILocator)this).Locator(GetByAltTextSelector(text, options?.Exact));
+
+    public ILocator GetByAltText(Regex text, LocatorGetByAltTextOptions options = null)
+        => ((ILocator)this).Locator(GetByAltTextSelector(text, options?.Exact));
+
+    public ILocator GetByLabel(string text, LocatorGetByLabelOptions options = null)
+        => ((ILocator)this).Locator(GetByLabelSelector(text, options?.Exact));
+
+    public ILocator GetByLabel(Regex text, LocatorGetByLabelOptions options = null)
+        => ((ILocator)this).Locator(GetByLabelSelector(text, options?.Exact));
+
+    public ILocator GetByPlaceholder(string text, LocatorGetByPlaceholderOptions options = null)
+        => ((ILocator)this).Locator(GetByPlaceholderSelector(text, options?.Exact));
+
+    public ILocator GetByPlaceholder(Regex text, LocatorGetByPlaceholderOptions options = null)
+        => ((ILocator)this).Locator(GetByPlaceholderSelector(text, options?.Exact));
+
+    public ILocator GetByRole(string role, LocatorGetByRoleOptions options = null)
+        => ((ILocator)this).Locator(GetByRoleSelector(role, new(options)));
+
+    public ILocator GetByTestId(string testId)
+        => ((ILocator)this).Locator(GetByTestIdSelector(testId));
+
+    public ILocator GetByText(string text, LocatorGetByTextOptions options = null)
+        => ((ILocator)this).Locator(GetByTextSelector(text, options?.Exact));
+
+    public ILocator GetByText(Regex text, LocatorGetByTextOptions options = null)
+        => ((ILocator)this).Locator(GetByTextSelector(text, options?.Exact));
+
+    public ILocator GetByTitle(string text, LocatorGetByTitleOptions options = null)
+        => ((ILocator)this).Locator(GetByTitleSelector(text, options?.Exact));
+
+    public ILocator GetByTitle(Regex text, LocatorGetByTitleOptions options = null)
+        => ((ILocator)this).Locator(GetByTitleSelector(text, options?.Exact));
+
+    internal static void SetTestIdAttribute(string attributeName)
+        => _testIdAttributeName = attributeName;
+
+    internal static string GetByTestIdSelector(string testId)
+           => GetByAttributeTextSelector(_testIdAttributeName, testId, exact: true);
+
+    internal static string GetByAttributeTextSelector(string attrName, string text, bool? exact)
+    {
+        var exactFlag = (exact == true) ? "s" : "i";
+        return $"internal:attr=[{attrName}={EscapeForAttributeSelector(text)}{exactFlag}]";
```

Comment:
Nit: you should follow up with upstream's exact as a part of escapeForAttributeSelector and also no 's' by default.